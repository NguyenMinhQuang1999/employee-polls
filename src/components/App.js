import { connect } from "react-redux";
import { Routes, Route } from "react-router-dom";
import LoadingBar from "react-redux-loading-bar";
import { handleInitialData } from "../actions/shared";
import { useEffect, Fragment } from "react";
import Login from "./Login";
import Nav from "./Nav";
import Home from "./Home";
import NewQuestion from "./NewQuestion";
import Leaderboard from "./Leaderboard";
import PageNotFound from "./PageNotFound";
import PollPage from "./PollPage";

const App = (props) => {
  useEffect(() => {
    props.dispatch(handleInitialData());
  }, []);

  console.log(props.authedUser, "props.authedUser");
 if (!props.authedUser) {
   return props.loading === true ? null : <Login />;
 }
  return (
    <div>
      <Fragment>
        <LoadingBar />
        {props.loading === true ? null : (
          <div className="container">
            <Nav />
            <Routes>
              <Route path="/" exact element={<Home />} />
              <Route path="/new" element={<NewQuestion />} />
              <Route path="/leaderboard" element={<Leaderboard />} />
              <Route path="/404" exact element={<PageNotFound />} />
              <Route path="*" exact element={<PageNotFound />} />
              <Route path="/question/:id" element={<PollPage />} />
            </Routes>
          </div>
        )}
      </Fragment>
    </div>
  );
};

const mapStateToProps = ({ authedUser, users }) => ({
  loading: !users,
  authedUser: authedUser,
});
export default connect(mapStateToProps)(App);
