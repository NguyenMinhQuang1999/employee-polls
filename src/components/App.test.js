import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import reducer from "../reducers";
import middleware from "../middleware";
import { createStore } from "redux";
import { setAuthedUser } from "../actions/authedUser";
import App from "./App";

const store = createStore(reducer, middleware);
describe("App", () => {
  it("should render the component", () => {
    // eslint-disable-next-line testing-library/render-result-naming-convention
    const component = render(
      <Provider store={store}>
        <BrowserRouter
          future={{
            v7_startTransition: true,
            v7_relativeSplatPath: true,
          }}
        >
          <App />
        </BrowserRouter>
      </Provider>
    );
    expect(component).toBeDefined();
    expect(component).toMatchSnapshot();
  });

});
