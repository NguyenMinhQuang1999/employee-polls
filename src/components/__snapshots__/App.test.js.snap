// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render the component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <div
          class="login-container"
        >
          <div
            class="logo"
          >
            <img
              alt="Logo"
              src="https://avatar.iran.liara.run/public"
            />
          </div>
          <h2>
            Log In
          </h2>
          <form>
            <div
              class="mb-3"
            >
              <label
                class="form-label"
                for="UserName"
              >
                User
              </label>
              <input
                class="form-control"
                data-testid="username"
                id="username"
                placeholder="User"
                type="text"
                value="mtsamis"
              />
            </div>
            <div
              class="mb-3"
            >
              <label
                class="form-label"
                for="password"
              >
                Password
              </label>
              <input
                class="form-control"
                data-testid="password"
                id="password"
                placeholder="Password"
                type="password"
                value="xyz123"
              />
            </div>
            <button
              class="btn btn-primary"
              data-testid="submit"
              type="submit"
            >
              Submit
            </button>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <div
        class="login-container"
      >
        <div
          class="logo"
        >
          <img
            alt="Logo"
            src="https://avatar.iran.liara.run/public"
          />
        </div>
        <h2>
          Log In
        </h2>
        <form>
          <div
            class="mb-3"
          >
            <label
              class="form-label"
              for="UserName"
            >
              User
            </label>
            <input
              class="form-control"
              data-testid="username"
              id="username"
              placeholder="User"
              type="text"
              value="mtsamis"
            />
          </div>
          <div
            class="mb-3"
          >
            <label
              class="form-label"
              for="password"
            >
              Password
            </label>
            <input
              class="form-control"
              data-testid="password"
              id="password"
              placeholder="Password"
              type="password"
              value="xyz123"
            />
          </div>
          <button
            class="btn btn-primary"
            data-testid="submit"
            type="submit"
          >
            Submit
          </button>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
